File: Sources/MutsumiPet/Models/PetMood.swift
import Foundation

enum PetMood: Int, CaseIterable, Equatable {
    case sleepy
    case curious
    case pleased

    var symbol: String {
        switch self {
        case .sleepy: "zzz"
        case .curious: "?"
        case .pleased: "♪"
        }
    }
}

File: windows/MutsumiPet.Windows/Models/PetMood.cs
namespace MutsumiPet.Models
{
    public enum PetMood
    {
        Sleepy = 0,
        Curious = 1,
        Pleased = 2
    }

    public static class PetMoods
    {
        public static readonly PetMood[] AllCases =
        {
            PetMood.Sleepy,
            PetMood.Curious,
            PetMood.Pleased
        };

        public static string Symbol(PetMood mood)
        {
            switch (mood)
            {
                case PetMood.Curious: return "?";
                case PetMood.Pleased: return "♪";
                default: return "zzz";
            }
        }
    }
}
